### ISSUE TITLE
Typo in GenericPullParser leader parsing?

### ISSUE THREAD

**Issue by dmolesUC** [ruby-marc/ruby-marc#85]
IRB just gave me the warning:

```none
…xml_parsers.rb:93: warning: found `= literal' in conditional, should be ==
```

The relevant [line of code](https://github.com/ruby-marc/ruby-marc/blob/02e944f06340814c3c65f8fbb33e17908dfdb634/lib/marc/xml_parsers.rb#L93):

```ruby
@current_element = nil if @current_element = :leader
```

It looks like this was introduced in 437b7ec.

My guess is that in practice `@current_element` is always already `:leader` when we hit this line of code, but I don't know for certain.

**Comment by jrochkind** [comment]
Although I'm not familiar with this code, that does seem like a bug -- the latter one really should be `==`. Would you like to submit a PR?

**Comment by jrochkind** [comment]
@cbeer, you made the the original commit (it looks like direct to master, not as part of a PR?), your feedback would be quite welcome!

It would be nice to have a test of the behavior introduced in 437b7ec, or a regression test for the bugfix here -- but I don't understand the situation that 437b7ec is about enough to write or suggest a test. If you had any hints for how to test 437b7ec, that'd be helpful!

**Comment by adjam** [comment]
This seems closely enough related to #87 that you could patch this (it clearly should be `==` , IMO) and roll it into that branch, but I'm not particularly fastidious about issue management.

**Comment by dmolesUC** [comment]
I went ahead and created a separate PR #89 — either you can merge it first, or I can rebase it after #87's merged. Or if the consensus is to just roll the fix into #87 that's OK too.
